### PR TITLE
Refactoring drakrun

### PR DESCRIPTION
**Pull request by psrok1**
Targets:

- Introduce Karton 5.0.0+ coding standards and compatibility
- Right now:  `drakrun` is not just an analysis runner. Actually this package contains whole toolkit including draksetup, regression tester and everything that need to run on machine with Xen and Drakvuf installed. We need to untangle dependencies and split things in `drakrun` to make it easier to maintain.

What I've done:
- Divided drakrun into subpackages:
    - `drakrun.machinery` for everything that is needed to set up a VM and provides Python interface to OS, Xen and Drakvuf tools
    - `drakrun.profile` for VMI profile stuff
    - `drakrun.utilities` for tools that are not related with Drakrun Karton itself, but are needed for setup and test suite
- Big "no no" for rewriting configurations on the fly, things should be consolidated on setup level
- Simplify packaging that is too generalized

### CODE REVIEW

File: drakrun/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,4 @@
-karton-core==4.3.0
-minio==5.0.7
-redis==4.4.4
```
Comment (msm-code) [inline drakrun/requirements.txt:3]: is it that easy? were they just unused dependencies? Or just transitive dependencies from karton-core?

**Review by msm-code — APPROVED** [review]
Weak approve, but I believe one LoC is broken now after refactor of Drakrun constructor.

File: drakrun/drakrun/main.py [hunk 2]
```diff
@@ -870,9 +851,9 @@ def main(args):
 
     # Apply Karton configuration overrides
     drakrun_conf = conf.config["drakrun"] if conf.config.has_section("drakrun") else {}
-    DrakrunKarton.reconfigure(drakrun_conf)
+    Drakrun.reconfigure(drakrun_conf)
 
-    c = DrakrunKarton(conf, args.instance)
+    c = Drakrun(conf, args.instance)
```
[the remark this answers is not in the document]
Comment (psrok1) [reply drakrun/drakrun/main.py]: Yes, at the time of review I've not rewritten that part yet

File: drakrun/drakrun/main.py [hunk 3]
```diff
@@ -134,56 +138,83 @@ class DrakrunKarton(Karton):
...
+
+        # Preferred plugins for specific feed quality
+        # [drakvuf_plugins]
+        # low = apimon,procmon...
+        self.active_plugins = {
+            "_all_": DEFAULT_DRAKVUF_PLUGINS
+        }
+        if self.config.has_section("drakvuf_plugins"):
+            for quality, list_str in self.config["drakvuf_plugins"].items():
+                plugins = [x for x in list_str.split(",") if x.strip()]
```
[the remark this answers is not in the document]
Comment (psrok1) [reply drakrun/drakrun/main.py:186]: Good point

File: drakrun/drakrun/main.py [hunk 4]
```diff
@@ -134,56 +138,83 @@ class DrakrunKarton(Karton):
...
-    DEFAULT_TEST_HEADERS = {
+    test_output_headers: Dict[str, str] = {
         "type": "analysis-test",
         "kind": "drakrun-internal",
     }
 
-    def __init__(self, config: Config, instance_id: int):
-        super().__init__(config)
+    def __init__(self, *args, instance_id: int, **kwargs):
+        super().__init__(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (psrok1) [reply drakrun/drakrun/main.py:148]: I thought about it and I'll probably do that later. That number on the front of `drakrun` command is too magic without some explanation.

**Comment by psrok1** [comment]
I finally decided to split things out of drakrun and make a separate `draklib` package. See #824.

**Pull request closed without merging**